src/main/java/seedu/address/logic/parser/LogCommandParser.java

```diff
@@ -40,23 +48,32 @@ public LogCommand parse(String args) throws ParseException {
                             LogCommand.MESSAGE_ONE_FLAG));
                 }
                 String flag = argMultimap.getValue(PREFIX_FLAG).get();
-                assert(flag.equals(LogCommand.FLAG_RESET) || flag.equals(LogCommand.FLAG_OVERRIDE));
+                assert(flag.equals(LogCommand.FLAG_RESET) || flag.equals(LogCommand.FLAG_OVERRIDE)
+                        || flag.equals(LogCommand.FLAG_RESET_ALL));
 
                 // Reset only depends on if a name is given,
                 // It should ignore any hour provided, and should not throw an error even if hour is null.
                 if (flag.equals("r")) {
-                    return new LogCommand(studySpot, null, isNamePresent,
-                            false, !isNamePresent);
+                    return new LogCommand(studySpot, null, isNamePresent, false, !isNamePresent);
+                }
+                else if (flag.equals("ra")) {
+                    return new LogCommand(studySpot, null, false, false, true);
                 } else {
                     isOverride = true;
                 }
             }
-
             hoursStudied = ParserUtil.parseStudiedHours(argMultimap.getValue(PREFIX_HOURS).get());
             return new LogCommand(studySpot, hoursStudied, false, isOverride, false);
         } catch (NoSuchElementException e) {
             throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, LogCommand.MESSAGE_USAGE));
         }
+    }
 
+    /**
+     * Returns true if none of the prefixes contains empty {@code Optional} values in the given
+     * {@code ArgumentMultimap}.
+     */
+    private static boolean arePrefixesPresent(ArgumentMultimap argumentMultimap, Prefix... prefixes) {
```

Comment:
Perhaps we can use `argMultimap.getValue(PREFIX_NAME).isPresent()` to retrieve name instead of creating a method as this method is only used once. Nevertheless, this could be written as a generic method in ParserUtil if many Parsers make use of this logic so that we avoid duplicate code across the different parsers.

Reply:
I have abstracted it out!